app.py

```diff
@@ -1,50 +1,150 @@
+
+socket = SocketIO(
+            app,
+            cors_allowed_origins='*',
+            message_queue=os.getenv('REDIS_URL'),
+            manage_session=False
+         ) 
+
+@app.template_filter('empty_session')
+def empty_session(session):
+    '''
+    This function removes all killed/timed-out jobs from the session if the video path for a 
+    given job does not exist. Afterwards, it specifies whether or not the session object is 
+    empty.
+    Parameters:
+    - session: Flask session object
+    '''
+   
+    if 'jobs' not in session:
+        logger.info('Empty session')
+        return True
+
+    def clear_session():
+        for idx, job_dict in enumerate(session['jobs']):
+            job = q.fetch_job(job_dict['id'])
+            video_source = 'static/videos/{}'.format(job_dict['tracks_filename'])
+            
+            dead_job_or_timeout = job is None or job.get_status() == 'failed'
+            video_does_not_exist = not os.path.exists(video_source)
+            
+            if dead_job_or_timeout and video_does_not_exist:
+                session['jobs'].pop(idx)
+                # Have to set modified to True after every in-place operation
+                session.modified = True 
+                logger.info('Popped job from session: id={}'.format(job_dict['id']))
+
+    clear_session()
+    
+    if len(session['jobs']) == 0:
+        logger.info('Empty session')
+        return True
+    
+    return False
 
-jobs = []
+@app.template_filter('refresh_job')
+def refresh_job(job_dict):
+    '''
+    This function updates the meta dictionary of a given job and returns the
+    job's filename
+    Parameters:
+    - job_dict: Dictionary containing a job id and job filename
+    '''
+    logger.info('Refresh job, id={}'.format(job_dict['id']))
+    job = q.fetch_job(job_dict['id'])
+    filename = job_dict['filename']
+    try:
+        job.refresh()
+        logger.info('Job refreshed successfully')
+    except Exception as e: 
+        logger.info('Job did not refresh properly, exception {}'.format(e))
+    return filename
 
+@app.template_filter('display_job')
+def display_job(job_dict):
+    '''
+    This function checks whether or not a video with the job's track filename 
+    exists. If it exists, then it returns HTML markup for the video source.
+    Else, it returns the progress information.
+    Parameters:
+    - job: Dictionary containing a job id and tracks filename
+    '''
+    logger.info('Displaying job, id={}'.format(job_dict['id']))
+    video_source = 'static/videos/{}'.format(job_dict['tracks_filename'])
+    if os.path.exists(video_source):
+        return Markup("""
+        <video width="320" height="240" controls>
+          <source src="/{}" type="video/mp4">
+        </video>
+        """.format(video_source))
+    job = q.fetch_job(job_dict['id'])
+    if job and job.meta.get('status'): 
+        return job.meta.get('status')
+    return 'Beginning process...'
+
+@app.route('/<path:path>')
```

Comment:
Please do not add a new route for every third party dependency. Just put the necesary files under the static path, e.g.:

- static
  - css
    - lib
      - spin.css
      - ...
    - ...
  - js
    - lib
      - spin.js
      - ...
    - ...
  - images
  - videos

Reply:
Done

Reply:
This route is far too open. You are making every file in the root directory available. What is the purpose of this route? If it is to serve static files, then please restrict it to the static folder.

Reply:
Done

Reply:
https://github.com/MLDSAI/VisionTracing/pull/2/commits/ea9b3c603148983d9b82e9b394d761a569d06efb

Reply:
https://github.com/MLDSAI/VisionTracing/pull/2/commits/ea9b3c603148983d9b82e9b394d761a569d06efb